Mathlib/RingTheory/Localization/Basic.lean

```diff
@@ -790,6 +790,18 @@ end IsLocalization
 
 section
 
+variable [IsLocalization M S] (M) (Q : Type*) [CommSemiring Q] {g : R →+* P} [Algebra P Q]
+
+theorem map_injective_of_injective (hg : Function.Injective g)
+    [IsLocalization (M.map g : Submonoid P) Q] :
+    Function.Injective (map Q g (Submonoid.le_comap_map M) : S → Q) :=
+  Submonoid.LocalizationMap.map_injective_of_injective (IsLocalization.toLocalizationMap M S) hg
+      (IsLocalization.toLocalizationMap (M.map g : Submonoid P) Q)
```

Comment:
If you're not planning to add more stuff to this section, you may remove the section and move the `variable` into the theorem statement.
```suggestion
theorem map_injective_of_injective (hg : Function.Injective g) [IsLocalization (M.map g) Q] :
    Function.Injective (map Q g (Submonoid.le_comap_map M) : S → Q) :=
  (toLocalizationMap M S).map_injective_of_injective hg (toLocalizationMap (M.map g) Q)
```